frontend/javascripts/oxalis/model/sagas/annotation_saga.tsx

```diff
@@ -234,29 +233,31 @@ export function* acquireAnnotationMutexMaybe(): Saga<void> {
           acquireAnnotationMutex,
           annotationId,
         );
-        yield* put(setAnnotationAllowUpdateAction(canEdit));
+        if (isInitialRequest && canEdit) {
+          // Only set allow update to true in case the first try to get the mutex succeeded.
+          yield* put(setAnnotationAllowUpdateAction(true));
+        } else {
+          yield* put(setAnnotationAllowUpdateAction(false));
+        }
```

Comment:
```suggestion
        // Only set allow update to true in case the first try to get the mutex succeeded.
        yield* put(setAnnotationAllowUpdateAction(isInitialRequest && canEdit));
```

Reply:
while my suggestion should be equal to the current code, I just noticed during testing that the logic itself is faulty. at the moment, on each iteration that is not the first, the boolean will be false and thus disabling the "is allowed" property. thus, I cannot edit my own annotation even when I got the mutex.

Reply:
> at the moment, on each iteration that is not the first, the boolean will be false and thus disabling the "is allowed" property. thus, I cannot edit my own annotation even when I got the mutex.

Ah, I see what the problem is. Sorry, that I missed that. I did not test manually for long enough for this to happen 😅 .

The code now remembers whether the mutex could be acquired from the beginning. Now the code only sets the "is allowed" to false, if the mutex could wasn't acquired from the beginning or `canEdit` sent by the server is false. This should be the correct logic right?

The new logic should now be:
=> Keep `allowUpdate` true in case the mutex was acquired initially and the server sends back `canEdit = true`
=> once the server sends `canEdit = false` `allowUpdate` will never be set to true again until the user reloads the page.
=> If the mutex wasn't acquired initially, the `allowUpdate` will always be false until the user reloads the page.

This should cover all cases now, right? And blocking all updates from the annotation once the mutex was lost only a single time should be correct, right? Because there could be another potential user that had the mutex in the meantime and made some small changes to the annotation.